security/s2a/internal/crypter/common.go

```diff
@@ -7,6 +7,12 @@ const (
 	// nonceSize is the size of the nonce in number of bytes for
 	// AES-128-GCM-SHA256, AES-256-GCM-SHA384, and CHACHA20-POLY1305-SHA256.
 	nonceSize = 12
+	// sha256DigestLength is the digest length of sha256 in bytes.
+	sha256DigestLength = 32
+	// sha384DigestLength is the digest length of sha384 in bytes.
+	sha384DigestLength = 48
+	// uint64Size is the size of a uint64 in bytes.
```

Comment:
Do we need this? Isn't this the definition of a `uint64`?

Reply:
We need this otherwise we'll have to use `8` as an arbitrary constant everywhere rather than `uint64Size`. We use this in the nonce mask function.

Reply:
But don't we already have a bunch of `8`'s in the `maskNonce` function anyway? And that seems to be the only place that `uint64Size` is used, unless I'm missing something.

If you want, you can add a comment in the `maskNonce` implementation that describes where the `8`'s are coming from?

Reply:
Oh those 8's were unintentional, those were meant to be replaced with `uint64Size`. I'll just remove the constant then and add a comment.

Reply:
On second thought, I just moved the constant over to the half connection file. I think having the constant is still useful.

Reply:
It doesn't need to be a global constant though. Can we have it be a local constant in the `maskedNonce` function?